spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/features/TaskConfiguration.java

```diff
@@ -102,10 +95,13 @@ public LauncherInitializationService launcherInitializationService(
 		return new LauncherInitializationService(launcherRepository, platforms);
 	}
 
+	@Profile({"local", "default"})
```

Comment:
what is the `default` profile?

Reply:
I removed "default" and all the tests still passed, prob was some cruft left over.

Reply:
TLDR; The default profile is active when no other profile is set.  https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-external-config.html#boot-features-external-config-profile-specific-properties
SCDF sets the profile to `local` via the `ProfileApplicationListener`, but since this is registered in `spring.factories` some of the Local*Tests were failing since the `local` profile was not set. Rather than explicitly setting the profile for the tests, which may cause future tests to fail if they forget to do that, and mislead some to think it must be explicitly set,  I decided it was safe to default to the local TaskPlatform if no other profile is set.

Reply:
yea, I just bumped into the other failing tests....